File: apps/frontend/src/lib/services/chatTopics.ts
// Contributor(s): Robbie
// Main: Robbie - SvelteKit fetch helpers for staff tools auth chat and shared API constants.

/**
 * WebSocket topic paths for chat. Keep in sync with ChatTopics.java on Android
 * and the topic chat thread strings configured on the Spring broker.
 */

const base = (threadId: number | string) => `/topic/chat/thread/${threadId}`;

export const ChatTopics = {
	messages: (threadId: number | string) => `${base(threadId)}/messages`,
	staffMessages: (threadId: number | string) => `${base(threadId)}/staff-messages`,
	typing: (threadId: number | string) => `${base(threadId)}/typing`,
	read: (threadId: number | string) => `${base(threadId)}/read`,
	status: (threadId: number | string) => `${base(threadId)}/status`,
	newThreads: () => '/topic/chat/threads',
	typingPublish: (threadId: number | string) => `/app/chat/thread/${threadId}/typing`
};

File: apps/backend/src/main/java/com/sait/peelin/service/ChatTopics.java
// Contributor(s): Robbie
// Main: Robbie - STOMP topic path helpers for chat typing and message fan-out.

package com.sait.peelin.service;

/**
 * Canonical WebSocket topic paths for chat. Kept in sync with the TS + Kotlin
 * equivalents in the web and Android clients.
 */
public final class ChatTopics {

    private ChatTopics() {}

    public static final String NEW_THREADS = "/topic/chat/threads";

    private static String base(Integer threadId) {
        return "/topic/chat/thread/" + threadId;
    }

    public static String messages(Integer threadId) {
        return base(threadId) + "/messages";
    }

    public static String staffMessages(Integer threadId) {
        return base(threadId) + "/staff-messages";
    }

    public static String typing(Integer threadId) {
        return base(threadId) + "/typing";
    }

    public static String read(Integer threadId) {
        return base(threadId) + "/read";
    }

    public static String status(Integer threadId) {
        return base(threadId) + "/status";
    }
}
